src/api/bm-response.ts

```diff
@@ -0,0 +1,50 @@
+import createInternal from './utils/internal'
+
+import type { Headers } from './types'
+import { APITypes } from '@oneblink/types'
+
+const internal = createInternal()
+
+class BmResponseImplementation<T>
+  implements APITypes.OneBlinkAPIHostingResponse<T> {
+  constructor() {
+    Object.assign(internal(this), {
+      headers: {},
+      payload: undefined,
+      statusCode: 200,
+    })
+  }
+
+  get headers(): Headers {
+    return Object.assign({}, internal(this).headers)
+  }
+
+  get payload(): any {
```

Comment:
Any chance of having this as `unknown` ? or will it be a giant pain?

Reply:
This should be `T` I think. Ill have a look.